zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -121,15 +137,43 @@ private void parseZNodeMapping() {
           "ZkClientUriDomainMappingHelper::parseZNodeMapping(): No application domain ZNodes found in root path: {}",
           rootPath);
     }
+    clientUriToDomainNames = newClientUriToDomainNames;
   }
 
   @Override
   public void process(WatchedEvent event) {
     parseZNodeMapping();
+    // Update AuthInfo for all the known connections.
+    ServerCnxnFactory factory =
```

Comment:
Correct. Technically, we're only interested in the clients that connect to `secureClientPort`, so just looking at `SecureServerCnxnFactory` should suffice?

Reply:
At least in the test, we are not using a secure factory. We will need to re-design some of the test logic if we want to restrict this. And I don't see much benefit of differentiating these 2 types. Let me know if I missed anything.

Reply:
This doesn't break anything, so it doesn't hurt. But can we do a `null` check for the regular serverCnxnFactory as well?

Let's clarify and add a TODO here. This technically isn't correct because we should only look at secure server cnxn factory. Ultimately, we want to deprecate all plaintext access.

Reply:
Sure, I guess it mainly prevents failures from test logic. But it avoids NPE and we should do it.